daemon/create_windows.go

```diff
@@ -2,20 +2,41 @@ package daemon
 
 import (
 	"fmt"
+	"strings"
 
 	containertypes "github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/container"
 	"github.com/docker/docker/pkg/stringid"
 	"github.com/docker/docker/volume"
 )
 
+// dedupConfigEnvVars deduplicates requested environment variables in the config
+// as they are case insensitive on Windows
+func dedupConfigEnvVars(config *containertypes.Config) {
+	deduped := make(map[string]string)
+	for _, v := range config.Env {
+		i := strings.Index(v, "=")
+		if i >= 0 {
+			name := strings.ToUpper(v[:i])
+			value := v[i+1:]
+			deduped[name] = value
+		}
+	}
+	config.Env = nil
+	for k, v := range deduped {
+		config.Env = append(config.Env, k+"="+v)
+	}
+}
```

Comment:
Actually, I'm going to close this one for other reasons which will become more obvious in the next few months. I'm not sure Windows is going to change this case insensitivity or have different shell behaviour any time soon, if ever (but it's actually part of the internal process structure more than the shell).